torch/lib/c10d/TCPStore.cpp

```diff
@@ -67,12 +89,30 @@ void TCPStoreDaemon::run() {
       fds[i].revents = 0;
     }
 
+#ifdef _WIN32
+    int res;
+    SYSCHECK_ERR_RETURN_NEG1(
+      res = WSAPoll(fds.data(), fds.size(), checkTimeout_.count()))
+    if (res == 0) {
+      auto rv = WaitForSingleObject(ghStopEvent_, 0);
+      if (rv != WAIT_TIMEOUT) {
+          finished = true;
+          break;
+      }
+      continue;
+    }
+#else
     SYSCHECK_ERR_RETURN_NEG1(::poll(fds.data(), fds.size(), -1));
+#endif
 
     // TCPStore's listening socket has an event and it should now be able to
     // accept new connections.
     if (fds[0].revents != 0) {
+#ifdef _WIN32
+      if (!(fds[0].revents & POLLIN)) {
+#else
       if (fds[0].revents ^ POLLIN) {
```

Comment:
can this condition be an inline function taking a const reference to `fds[0]`, and then can be implemented differently in `Win/UnixSocketUtil.hpp`?